### ISSUE TITLE
Associations guide incorrectly states belongs_to as one-to-one relationship

### ISSUE THREAD

**Issue by aspiers** [rails/rails#37899]
https://edgeguides.rubyonrails.org/association_basics.html#the-belongs-to-association says:

> A `belongs_to` association sets up a one-to-one connection with another model

However (unless I'm going crazy), this is not accurate, because `belongs_to` can also set up a many-to-one relationship as the other side of `has_many`.  This is explicitly stated in [the `has_many` section of the same guide](https://edgeguides.rubyonrails.org/association_basics.html#the-has-many-association):

> A `has_many` association indicates a one-to-many connection with another model. You'll often find this association on the "other side" of a `belongs_to` association.

So `belongs_to` is actually a "one-to-something" relationship, where *something* can either be *one* or *many*.

A knock-on problem with this wording is found in the beginning of [the `has_one` section](https://edgeguides.rubyonrails.org/association_basics.html#the-has-one-association), which says:

> A `has_one` association also sets up a one-to-one connection with another model, but with somewhat different semantics (and consequences). 

This sentence is clearly referring back to the previous `belongs_to` section, but "also" implies that they are both one-to-one, and this is not correct as per above.  This similarly invalidates the "but with ..." comparative clause.

So in summary I think the introductory paragraphs of both sections for `belongs_to` and `has_one` need to be reworded.

**Comment by wbhouston** [comment]
I agree that the wording is confusing, and doesn't line up with [other places in the guide as well](https://edgeguides.rubyonrails.org/association_basics.html#belongs-to-association-reference). A [normal practice](https://edgeguides.rubyonrails.org/association_basics.html#bi-directional-associations) is to have a `has_one` or `has_many` declaration for each `belongs_to` declaration, so it follows that the `belongs_to` sections should mention that.

In addition, I think none of the descriptions should be worded as the [belongs_to](https://edgeguides.rubyonrails.org/association_basics.html#the-belongs-to-association) and [has_one](https://edgeguides.rubyonrails.org/association_basics.html#the-has-one-association) sections are:

> A belongs_to association sets up a one-to-one connection with another model

The `belongs_to` declaration does not "set up" a one-to-one connection, it "indicates" a one-to-one connection (as the wording is in the [has_many](https://edgeguides.rubyonrails.org/association_basics.html#the-has-many-association) section). It is misleading to say that it sets up a one-to-one connection, when setting up a _valid_ one-to-one association involves much more than `belongs_to` and `has_one`. This is briefly mentioned at the end of the [has_one](https://edgeguides.rubyonrails.org/association_basics.html#the-has-one-association) section:

> Depending on the use case, you might also need to create a unique index and/or a foreign key constraint on the supplier column for the accounts table.

I think it would be useful to include a chapter here or in the [Validations Guide](https://edgeguides.rubyonrails.org/active_record_validations.html) the briefly explains how to validate a one-to-one association (and/or others) at the model-level.

**Comment by BKSpurgeon** [comment]
Ooops: I had not realised that yours was a pull request @Vasfed only just now did it click.

**Comment by woleywayzz** [comment]
The codre is scripted as to have vantage in protection , brute force is my specialty

**Comment by aspiers** [comment]
This is not stale, it's waiting for #37905 to be merged.
